src/dbnode/persist/types.go

```diff
@@ -136,19 +136,16 @@ type DataPrepareOptions struct {
 	NamespaceMetadata namespace.Metadata
 	BlockStart        time.Time
 	Shard             uint32
-	FileSetType       FileSetType
-	DeleteIfExists    bool
+	// This volume index is only used when preparing for a flush fileset type.
+	// When opening a snapshot, the new volume index is determined by looking
+	// at what files exist on disk.
```

Comment:
Again, does seem strange they don't just both do the same strategy.

Reply:
Yeah agreed its a little weird, and Justin's initial implementation did do the +1 strategy but we ran into some shard edges with that approach that pushed us towards the other direction